File: prete_service/src/main/java/prete_service/DTO/Livre.java
package prete_service.DTO;

import lombok.*;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
public class Livre {
    private Integer idLivre;
    private String titre;
    private String auteur;
    private String genre;
    private String isbn;
    private Integer numChapters;
    private Integer numPages;
    private Integer numTotalLivres;
}

File: livre_service/src/models/Livre.js
const { DataTypes } = require("sequelize");
const sequelize = require("../config/database");

const Livre = sequelize.define("Livre", {
  idLivre: {
    type: DataTypes.INTEGER,
    primaryKey: true,
    autoIncrement: true
  },
  titre: {
    type: DataTypes.STRING,
    allowNull: false
  },
  isbn: {
    type: DataTypes.STRING,
    unique: true 
  },
  auteur: {
    type: DataTypes.STRING
  },
  genre: {
    type: DataTypes.STRING
  },
  numChapters: {
    type: DataTypes.INTEGER
  },
  numPages: {
    type: DataTypes.INTEGER
  },
  numTotalLivres: {
    type: DataTypes.INTEGER
  },
   image: {
  type: DataTypes.BLOB // BYTEA en PostgreSQL
},
  synopsis: {                 
    type: DataTypes.TEXT
  }
});

module.exports = Livre;
